src/scripts/quests/questTypes/GainTokensQuest.ts

```diff
@@ -8,13 +8,17 @@ class GainTokensQuest extends Quest implements QuestInterface {
     }
 
     public static generateData(): any[] {
-        const amount = SeededRand.intBetween(1000, 8000);
-        const reward = this.calcReward(amount);
-        return [amount, reward];
+        const regionDungeons = GameConstants.RegionDungeons[player.highestRegion()].filter(dungeon => TownList[dungeon].isUnlocked());
+        const maxDungeon = dungeonList[regionDungeons[regionDungeons.length - 1]];
```

Comment:
Similar issue with the maxGym calc - it's possible for no dungeons to be unlocked